system/HTTP/SiteURI.php

```diff
@@ -145,7 +145,7 @@ private function determineBaseURL(
         $uri = new URI($baseURL);
 
         // Update scheme
-        if ($scheme !== null) {
+        if ($scheme !== null && $scheme !== '') {
```

Comment:
How about:
```suggestion
        if ($scheme) {
```

Reply:
@MGatner truthy check should only for bool value, using exact compare null or empty string is better to know the exact usage